### PR TITLE
Add rewards data and fix polygon issue

### PR DESCRIPTION
**Pull request by termaatFlipside**

### CODE REVIEW

File: app/routes/app/rewards/addresses.tsx [hunk 1]
```diff
@@ -53,20 +54,21 @@ export async function action({ request }: ActionArgs) {
 }
 
 export const loader = async (data: DataFunctionArgs) => {
-  const user = await getUserId(data.request);
-  const wallets = user ? await findAllWalletsForUser(user) : [];
-  const tokens = await listTokens();
+  const user = await getUser(data.request);
+  invariant(user);
```
Comment (sebruiz) [inline app/routes/app/rewards/addresses.tsx:58]: Add an error message here to make it easier to track

File: app/routes/app/rewards/addresses.tsx [hunk 2]
```diff
@@ -53,20 +54,21 @@ export async function action({ request }: ActionArgs) {
...
+  const wallets = await findAllWalletsForUser(user.id);
+  const rewards = await searchUserSubmissions(user.address);
```
Comment (sebruiz) [inline app/routes/app/rewards/addresses.tsx:60]: might be nitpicky: these aren't really rewards. They're still submissions

File: app/services/submissions.server.ts [hunk 3]
```diff
@@ -129,3 +129,10 @@ export const prepareSubmission = async (
   });
   return contractData;
 };
+
+/**
+ * Returns an array of SubmissionDoc for a given UserAddress
+ */
+export const searchUserSubmissions = async (address: string) => {
```
Comment (sebruiz) [inline app/services/submissions.server.ts:136]: I think we can consolidate with `searchSubmissions` if we add serviceProvider as one of the search params

File: app/routes/app/rewards/index.tsx [hunk 4]
```diff
@@ -27,25 +27,29 @@ import { ClaimRewardWeb3Button } from "~/features/web3-button/claim-reward";
...
+import { searchSubmissions } from "~/services/submissions.server";
+import type { SubmissionDoc } from "~/domain/submission";
 
 export const loader = async (data: DataFunctionArgs) => {
-  const user = await getUserId(data.request);
-  const wallets = user ? await findAllWalletsForUser(user) : [];
+  const user = await getUser(data.request);
+  invariant(user, "Could not find user, please sign in");
+  const wallets = await findAllWalletsForUser(user.id);
+  const submissions = await searchSubmissions({
```
Comment (sebruiz) [inline app/routes/app/rewards/index.tsx:37]: I think you can get away with only passing in serviceProvider and the rest will default to a value right?

Comment (termaatFlipside) [reply app/routes/app/rewards/index.tsx:37]: That was givig me the red squiggle

Comment (termaatFlipside) [reply app/routes/app/rewards/index.tsx:37]: Also could just add filter sort to this branch since doing that would mean these get passed in anyway

Comment (sebruiz) [reply app/routes/app/rewards/index.tsx:37]: do it!

**Review by sebruiz — APPROVED** [review]

**Pull request merged**